docs/proposals/PROPOSAL_B.md

```diff
@@ -0,0 +1,405 @@
+
+##### Variant A
+
+A completely new language to present image compatibility as a graph.
+
+- **`spec`** *string-object map*
+
+  This REQUIRED property describes the compatibility specification.
+
+  - **`<compatibilityGroup>`** *object*
+
+    This REQUIRED property specifies the compatibility group (a DAG node).
+
+    - **`compatibilities`** *string-object*
+
+      This OPTIONAL property is a list of compatibility domains.
+
+      - **`<compatibilityDomain>`** *object*
+
+        This REQUIRED property specifies the compatibility domain. Domain names are used to identify origin of data and its funcionality. The name must be a valid domain name.
+
+        - **`annotations`** *string-string map*
+
+          This OPTIONAL property specifies optional annotations.
+
+        - **`requirements`** *object*
+
+          This REQUIRED property describes data that reflects container requirements against the host OS. Allowed fields must be described in the documentation of the compatibility domain.
+
+    - **`relation`** *object*
+
+      This OPTIONAL property specifies the relation to other compatibility groups (nodes). Only one property is allowed _all_, _oneOf_ or _noneOf_.
+
+      - **`all`** *array of strings*
+
+        This OPTIONAL property specifies a list of all required compatibility groups that must pass that container is considered to be compatible with a host.
+
+      - **`oneOf`** *array of strings*
+
+        This OPTIONAL property specifies a list of compatibility groups with a condition that at least one group has to pass that container is considered to be compatible with a host.
+
+      - **`noneOf`** *array of strings*
+
+        This OPTIONAL property specifies a list of compatibility groups that cannot pass that container is considered to be compatible with a host.
+
+Example of use
+
+```json
+{
+  "spec": {
+    "intelCpu": {
+      "compatibilities": {
+        "org.opencontainers.hardware.cpu": {
+          "requirements": {
+            "vendor": "GenuineIntel",
+            "virtualization": "VT-x"
+          }
+        }
+      },
+      "relation": {
+        "all": ["intelIommu"]
+      }
+    },
+    "intelIommu": {
+      "compatibilities": {
+        "org.opencontainers.kernel.cmdline": {
+          "requirements": {
+            "intel_iommu": "on"
+          }
+        }
+      },
+      "relation": {
+        "all": ["vfio"]
+      }
+    },
+    "amdCpu": {
+      "compatibilities": {
+        "org.opencontainers.hardware.cpu": {
+          "requirements": {
+            "vendor": "GenuineIntel",
+            "virtualization": "AMD-V"
+          }
+        }
+      },
+      "relation": {
+        "all": ["amdIommu"]
+      }
+    },
+    "amdIommu": {
+      "compatibilities": {
+        "org.opencontainers.kernel.cmdline": {
+          "requirements": {
+            "amd_iommu": "pt"
+          }
+        }
+      },
+      "relation": {
+        "all": ["vfio"]
```

Comment:
something can not work well here:

for example: 
case 1:  intel CPU only work with intel GPU, and AMD CPU only work with AMD GPU. because vfio node is shared both for intel IOMMU and amd IOMMU, that will force intel CPU work not only for intel GPU but also for AMD GPU because vfio node has relation to intel GPU and AMD GPU

case 2:  intel CPU work both intel GPU/AMD GPU, amd CPU only work with AMD GPU. because vfio node is shared both for intel IOMMU and amd IOMMU, that will force AMD CPU work not only for intel GPU but also for AMD GPU because vfio node has relation to intel GPU and AMD GPU, the vfio node here are OK with intel CPU.

Reply:
The graph of container requirements I represented requires:
1. Intel CPU with Intel or NVIDIA GPU
2. AMD CPU with Intel or NVIDIA GPU

That's why VFIO node is shared.

It is just an example of the container image with specific requirements.
If any container image would have any other specific requirements then of course other combinations are possible. However, I didn't want to expand the graph to make it simple for the proposal.

As mentioned in here https://github.com/opencontainers/wg-image-compatibility/pull/6/files#diff-644786f78aa35025b6d4493a27fcc3d34c7a50df73c81c135daff93025a620c8R123-R129